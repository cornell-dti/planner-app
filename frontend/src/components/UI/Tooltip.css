.tooltip {
  position: relative;
  display: inline-block;
  border-bottom: 1px dotted black;
}

.tooltip .tooltiptext {
  width: 120px;
  background-color: black;
  color: #fff;
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;
  position: absolute;
  z-index: 1;
  top: 100%;
  left: 50%;
  margin-left: -100px;
  margin-top: -20px;
  font-size: 1.25em;
}

.tooltip .tooltiptext::after {
  content: " ";
  position: absolute;
  bottom: 100%;
  left: 50%;
  margin-left: 34px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent transparent black transparent;
}
