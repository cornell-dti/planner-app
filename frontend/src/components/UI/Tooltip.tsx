import React, { Component, ReactElement } from 'react';
import SamwiseIcon from './SamwiseIcon';
import { IconName } from './samwise-icon-types';
import styles from './Tooltip.module.css';

type Props = {
  readonly text: string;
  readonly iconName: IconName;
}
type State = {
  hover: boolean;
}

export default class Tooltip extends Component<Props, State> {
  public readonly state: State = { hover: false };

  private handleMouseIn = () => this.setState({ hover: true });

  private handleMouseOut = () => this.setState({ hover: false });

  public render(): ReactElement {
    const { text, iconName } = this.props;
    const { hover } = this.state;
    const tooltipStyle = { display: hover ? 'block' : 'none' };

    return (
      <div>
        <div
          onMouseOver={this.handleMouseIn}
          onMouseOut={this.handleMouseOut}
          onFocus={this.handleMouseIn}
          onBlur={this.handleMouseOut}
        >
          <SamwiseIcon iconName={iconName} />
        </div>
        <div className={styles.tooltip}>
          <div className={styles.tooltiptext} style={tooltipStyle}>{text}</div>
        </div>
      </div>
    );
  }
}
