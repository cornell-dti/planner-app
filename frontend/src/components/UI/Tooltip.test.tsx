import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './Tooltip';

it('Tooltip matches snapshot', () => {
  const tree = renderer.create(<Tooltip text="text" iconName="alert" />).toJSON();
  expect(tree).toMatchSnapshot();
});
