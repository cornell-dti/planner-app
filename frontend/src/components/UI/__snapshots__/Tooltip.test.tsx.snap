// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip matches snapshot 1`] = `
<div>
  <div
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <span
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
      title="alert"
    >
      <img
        alt="alert"
        src="/svgs/alert.svg"
        style={
          Object {
            "height": "1em",
            "width": "1em",
          }
        }
      />
    </span>
  </div>
  <div
    className="tooltip"
  >
    <div
      className="tooltiptext"
      style={
        Object {
          "display": "none",
        }
      }
    >
      text
    </div>
  </div>
</div>
`;
